tools/statplot.py

```diff
@@ -237,50 +272,74 @@ def PlotData(self, action, type, xscale, yscale):
         self.fig.canvas.draw()
         self.fig.canvas.flush_events()
 
-    def PopulateCombo(self, action, prevIterX=None, prevIterY=None):
-        for entry in sorted(self.entries.Paths()):
-            self.xCombo.append_text(entry)
-            self.yCombo.append_text(entry)
-        if action == 'load':
-            if 'ElapsedTime' in self.entries.Paths():
-                iterX = self.xCombo.get_model(). \
-                    get_iter(sorted(self.entries.Paths()).index('ElapsedTime'))
-                self.xCombo.set_active_iter(iterX)
-                self.yCombo.set_active(0)
-            else:
-                self.xCombo.set_active(0)
-                self.yCombo.set_active(1)
-        elif action == 'reload':
-            self.xCombo.set_active_iter(prevIterX)
-            self.yCombo.set_active_iter(prevIterY)
+    def ReadData(self):
+        def GatherEntries(stat2read):
+            print('Reading ' + stat2read)
+            entries = numpy.array([])
```

Comment:
I think this might be clearer as a regular Python list, then later on we'd only need

```python
entries.append(...)
```

To support the argsort indexing later, it can be converted to a numpy array from the list

Reply:
Agree the Numpy append syntax is quite verbose; fixed.